src/js/controller/controller.js

```diff
@@ -728,22 +733,23 @@ define([
             this.createInstream = function() {
                 this.instreamDestroy();
                 this._instreamAdapter = new InstreamAdapter(this, _model, _view);
-                // Persist the current captions track so the index
-                // is  set correctly after ad playback ends
-                _model.persistCaptionsTrack();
+                _model.set('adState', this.getAdState());
```

Comment:
Instream has a reference to the model and should update the model with it's state when it's changes.